libs/vm_modules/tests/unit/state_tests.cpp

```diff
@@ -259,18 +259,14 @@ TEST_F(StateTests, test_serialisation_of_complex_type)
 
   ASSERT_TRUE(toolkit.Compile(deser_src));
 
-  // TODO (issue 1172): This expectation is just temporary, it shall be replaced
-  // with commented-out code bellow once the issue is resolved.
   try
```

Comment:
All of this try-catch block & commented-out code bellow shall be replaced by code bellow (which however is failing in runtime - there is still an outstanding issue with deserialisation of Array of objects):

```suggestion
  Variant output;
  ASSERT_TRUE(toolkit.Run(&output));
  ASSERT_FALSE(output.IsPrimitive());

  auto retval{output.Get<Ptr<IArray>>()};
  ASSERT_TRUE(static_cast<bool>(retval));
  ASSERT_EQ(int32_t{3}, retval->Count());
  EXPECT_EQ(std::string{"aaa"}, retval->PopFrontOne().Get<Ptr<String>>()->str);
  EXPECT_EQ(std::string{"bbb"}, retval->PopFrontOne().Get<Ptr<String>>()->str);
  EXPECT_EQ(std::string{"ccc"}, retval->PopFrontOne().Get<Ptr<String>>()->str);
}
```